### PR TITLE
Support for Open interest interval in Binance Futures

### PR DESCRIPTION
**Pull request by jinusean**
Allows polling interval for open interest in Binance Futures. 

- Fixed some syntax errors from previous merge. 
- Fixed HTTPConcurrentPoll to handle reconnects.

- [x] - Tested
- [x] - Changelog updated
- [x] - Tests run and pass
- [x] - Flake8 run and all errors/warnings resolved
- [ ] - Contributors file updated (optional)

### CODE REVIEW

File: cryptofeed/connection.py [hunk 1]
```diff
@@ -230,30 +231,28 @@ class HTTPConcurrentPoll(HTTPPoll):
...
+        tasks = asyncio.gather(*(self._poll_address(address, header) for address in self.address))
 
-        while True:
-            data = await self._queue.get()
-            if isinstance(data, Exception):
-                raise data
-            yield data
+        try:
+            while not tasks.done():
+                with suppress(asyncio.exceptions.TimeoutError):
```
Comment (bmoscon) [inline cryptofeed/connection.py:247]: can you explain why this block of code is necessary?

File: cryptofeed/exchanges/binance.py [hunk 2]
```diff
@@ -454,5 +454,7 @@ async def subscribe(self, conn: AsyncConnection):
         # Binance does not have a separate subscribe message, the
         # subscription information is included in the
         # connection endpoint
-        if not isinstance(conn, HTTPPoll):
+        if isinstance(conn, (HTTPPoll, HTTPConcurrentPoll)):
+            self.open_interest = {}
```
Comment (bmoscon) [inline cryptofeed/exchanges/binance.py:458]: there isnt currently a defined `open_interest` method in the RestExchange class or the RestMixins, but there probably will be soon. Naming should be changed to _open_interest. I know this was here before you modified it, but now is as good a time as any to change it

**Review by bmoscon — CHANGES_REQUESTED** [review]
see previous comments

File: cryptofeed/exchanges/binance.py (hunk 2 above)
Comment (bmoscon) [reply cryptofeed/exchanges/binance.py:458]: this no longer applies, I changed the new to self._open_interest_cache, so just update the name here

**Comment by bmoscon** [comment]
@jinusean are you able to make the requested changes?

**Comment by jinusean** [comment]
Yes I've just pushed the relevant changes.

**Review by bmoscon — APPROVED** [review]

**Pull request merged**